components/authz-service/server/project.go

```diff
@@ -811,35 +811,42 @@ func fromAPIType(t api.ProjectRuleTypes) (storage.RuleType, error) {
 	}
 }
 
-func (s *ProjectState) prepareStorageRule(inID, projectID, name string,
-	inType api.ProjectRuleTypes, inConditions []*api.Condition) (*storage.Rule, error) {
+type apiRule interface {
```

Comment:
Q: Couldn't we re-use the protobuf-generated type here?

Reply:
I wanted a type that fit both the Create and Update request types.